crates/common/src/provider/runtime_transport.rs

```diff
@@ -211,19 +211,31 @@ impl RuntimeTransport {
     pub fn request(&self, req: RequestPacket) -> TransportFut<'static> {
         let this = self.clone();
         Box::pin(async move {
-            if this.inner.read().await.is_none() {
-                let mut inner = this.inner.write().await;
-                *inner = Some(this.connect().await.map_err(TransportErrorKind::custom)?)
+            let mut inner = this.inner.read().await;
+            if inner.is_none() {
+                drop(inner);
+                let mut inner_mut = this.inner.write().await;
+                if inner_mut.is_none() {
+                    *inner_mut = Some(this.connect().await.map_err(TransportErrorKind::custom)?);
+                }
+                drop(inner_mut);
```

Comment:
pedantic style nit:

put this in its own scope {} so we don't need the manual drop